dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/teradata/QueryLogTableNamesResolver.java

```diff
@@ -0,0 +1,137 @@
+
+import static com.google.edwmigration.dumper.application.dumper.ConnectorArguments.OPT_QUERY_LOG_ALTERNATES;
+import static com.google.edwmigration.dumper.application.dumper.ConnectorArguments.OPT_QUERY_LOG_ALTERNATES_DEPRECATION_MESSAGE;
+import static com.google.edwmigration.dumper.application.dumper.connector.teradata.AbstractTeradataConnector.DEF_LOG_TABLE;
+import static com.google.edwmigration.dumper.application.dumper.connector.teradata.TeradataLogsConnector.ASSESSMENT_DEF_LOG_TABLE;
+import static com.google.edwmigration.dumper.application.dumper.connector.teradata.TeradataLogsConnector.TeradataLogsConnectorProperty.QUERY_LOGS_TABLE;
+import static com.google.edwmigration.dumper.application.dumper.connector.teradata.TeradataLogsConnector.TeradataLogsConnectorProperty.SQL_LOGS_TABLE;
+import static com.google.edwmigration.dumper.application.dumper.connector.teradata.TeradataUtils.nonEmpty;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+import com.google.edwmigration.dumper.application.dumper.ConnectorArguments;
+import com.google.edwmigration.dumper.application.dumper.MetadataDumperUsageException;
+import java.util.List;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Resolver that retrieves the names of query log tables from the command-line options.
+ *
+ * <p>Teradata saves query logs in two tables: {@code DBC.DBQLogTbl} and {@code DBC.DBQLSqlTbl}.
+ * These two tables are called query log table and SQL log table for the purposes of the query logs
+ * extraction.
+ *
+ * <p>Depending on the flags provided on the command-line, the {@code teradata-logs} connector uses
+ * different pair of source tables for extraction:
+ *
+ * <ul>
+ *   <li>If the {@code --assessment} flag is not used, then the logs are extracted from tables
+ *       {@code DBC.DBQLogTbl} and {@code DBC.DBQLSqlTbl}
+ *   <li>If the {@code --assessment} flag is used, then the logs are extracted from tables {@code
+ *       DBC.QryLogV} (a view that selects the logs from {@code DBC.DBQLogTbl} table) and {@code
+ *       DBC.DBQLSqlTbl}
+ * </ul>
+ *
+ * <p>Additionally, the user can specify an alternative table for each of the two tables:
+ *
+ * <ul>
+ *   <li>For the query log table, the command-line flag is: {@code -Dteradata-logs.query-log-table}
+ *   <li>For the SQL log table, the command-line flag is: {@code -Dteradata-logs.sql-log-table}
+ * </ul>
+ *
+ * <p>Deprecated: There is also a second way of specifying both alternative tables by using the
+ * {@code --query-log-alternates} command-line flag that accepts these two log tables as a parameter
+ * (separated by a comma).
+ */
+class QueryLogTableNamesResolver {
+  private static final Logger LOG = LoggerFactory.getLogger(QueryLogTableNamesResolver.class);
+
+  private static final String TERADATA_LOGS_CONNECTOR_NAME = "teradata-logs";
+
+  static QueryLogTableNames resolve(ConnectorArguments arguments) {
```

Comment:
optional: This method looks pretty complex, I would advise to split it into smaller functions. There are basically 2 separate resolve logics based on the `legacyAlternates` variable. You could extract it into two separate functions, both returning `QueryLogTableNames`.

Reply:
Done.